DMCompiler/DM/Visitors/DMVisitorExpression.cs

```diff
@@ -85,14 +89,20 @@ class DMVisitorExpression : DMASTVisitor {
                 Result = new Expressions.Args(identifier.Location);
             } else {
                 DMProc.DMLocalVariable localVar = _proc?.GetLocalVariable(name);
-
-                if (localVar != null) {
+                if (localVar != null && _scopeMode == "normal") {
                     Result = new Expressions.Local(identifier.Location, localVar.Type, name);
                     return;
                 }
 
+                int? procGlobalId = _proc.GetGlobalVariableId(name);
```

Comment:
This is throwing a `NullReferenceException` when `_proc` is null
```suggestion
                int? procGlobalId = _proc?.GetGlobalVariableId(name);
```